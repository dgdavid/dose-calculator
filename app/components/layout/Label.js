import React from 'react';
import PropTypes from 'prop-types';
import { Text, StyleSheet, View } from 'react-native';

import colors from '../../config/colors';

const Label = ({ text, textStyle, wrapperStyle, lineStyle, toUpperCase }) => (
  <View style={[styles.wrapper, wrapperStyle]}>
    <View style={[styles.line, lineStyle]} />
    <Text style={[styles.text, textStyle]}>
      { toUpperCase ? text.toUpperCase() : text }
    </Text>
  </View>
);

Label.propTypes = {
  text: PropTypes.string.isRequired,
  wrapperStyle: View.propTypes.style,
  textStyle: Text.propTypes.style,
  lineStyle: View.propTypes.style,
  toUpperCase: PropTypes.bool,
};

Label.defaultProps = {
  toUpperCase: true,
};

const styles = StyleSheet.create({
  wrapper: {
    width: '90%',
    height: 20,
    marginLeft: '5%',
  },

  line: {
    alignSelf: 'center',
    borderBottomColor: colors.darkGray,
    borderBottomWidth: 0.5,
    marginTop: 10,
    width: '90%',
  },

  text: {
    fontSize: 10,
    marginTop: -7.5,
    marginLeft: 0,
    paddingLeft: 7.5,
    paddingRight: 7.5,
    alignSelf: 'center',
    backgroundColor: colors.white,
  },
});

export default Label;
