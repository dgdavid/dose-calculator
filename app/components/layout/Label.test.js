import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Label from './Label';

describe('render correctly', () => {
  test('without props apart form required `text`', () => {
    const tree = renderer.create(<Label text="Label text in uppercases by default" />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('setting `toUpperCase` to false', () => {
    const tree = renderer.create(<Label text="Label text rendered as is" toUpperCase={false} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
