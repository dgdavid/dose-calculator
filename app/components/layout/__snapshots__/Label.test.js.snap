// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly setting \`toUpperCase\` to false 1`] = `
<View
  style={
    Array [
      Object {
        "height": 20,
        "marginLeft": "5%",
        "width": "90%",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "borderBottomColor": "#A9A9A9",
          "borderBottomWidth": 0.5,
          "marginTop": 10,
          "width": "90%",
        },
        undefined,
      ]
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "alignSelf": "center",
          "backgroundColor": "#FFF",
          "fontSize": 10,
          "marginLeft": 0,
          "marginTop": -7.5,
          "paddingLeft": 7.5,
          "paddingRight": 7.5,
        },
        undefined,
      ]
    }
  >
    Label text rendered as is
  </Text>
</View>
`;

exports[`render correctly without props apart form required \`text\` 1`] = `
<View
  style={
    Array [
      Object {
        "height": 20,
        "marginLeft": "5%",
        "width": "90%",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "borderBottomColor": "#A9A9A9",
          "borderBottomWidth": 0.5,
          "marginTop": 10,
          "width": "90%",
        },
        undefined,
      ]
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "alignSelf": "center",
          "backgroundColor": "#FFF",
          "fontSize": 10,
          "marginLeft": 0,
          "marginTop": -7.5,
          "paddingLeft": 7.5,
          "paddingRight": 7.5,
        },
        undefined,
      ]
    }
  >
    LABEL TEXT IN UPPERCASES BY DEFAULT
  </Text>
</View>
`;
